deployment/ccip/changeset/cs_deploy_chain.go

```diff
@@ -605,55 +624,51 @@ func deployChainContractsSolana(
 			return fmt.Errorf("failed to save address: %w", err)
 		}
 	} else {
-		e.Logger.Infow("Using existing router", "addr", chainState.SolCcipRouter.String())
-		ccipRouterProgram = chainState.SolCcipRouter
+		e.Logger.Infow("Using existing router", "addr", chainState.Router.String())
+		ccipRouterProgram = chainState.Router
 	}
-	ccip_router.SetProgramID(ccipRouterProgram)
+	solRouter.SetProgramID(ccipRouterProgram)
 
 	// check if solana router is initialised
-	initialized, err := checkRouterInitialized(e, chain, ccipRouterProgram)
+	var routerConfigAccount solRouter.Config
+	err = chain.GetAccountDataBorshInto(e.GetContext(), GetRouterConfigPDA(ccipRouterProgram), &routerConfigAccount)
 	if err != nil {
-		return err
-	}
-	if initialized {
+		if err := initialzeRouter(e, chain, ccipRouterProgram); err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if err2 := initialzeRouter(e, chain, ccipRouterProgram); err2 != nil {
			return fmt.Errorf("failed to initialize router: %w", err2)
		}
```
We should have a linting rule against this kind of shadowing, not sure why its not firing here.